application/src/main/java/org/togetherjava/tjbot/commands/basic/SuggestionsUpDownVoter.java

```diff
@@ -46,6 +46,10 @@ public void onMessageReceived(@NotNull MessageReceivedEvent event) {
 
         Guild guild = event.getGuild();
         Message message = event.getMessage();
+        
+        if (!message.getChannel().getType().isThread()) {
```

Comment:
This `if` should not only affect thread creation but also the emoji-reactions. All in all, it should basically affect the whole method. Please move it up into the general checks at the top, i.e. the initial `if` that `return`s.

Reply:
ok 👍🏻

Reply:
```java
        if ((event.getAuthor().isBot() || event.isWebhookMessage() || !event.isFromGuild()) && !event.getChannel().getType().isThread()) {
            return;
        }
```
like this?

Reply:
got it
```java
        if (event.getAuthor().isBot() || event.isWebhookMessage() || !event.isFromGuild() || event.getChannel().getType().isThread()) {
            return;
        }
```

Reply:
Better, but still not exactly what I was looking for. You are still checking whether the message is a thread-message. But instead, you should just look for "not ordinary message". I.e. the code should only react on ordinary text messages, not on any other fancy type. Right now you only exclude thread-types but what about all other unwanted types.

Reply:
i don't see any other type of message is suggestions channel there is only the actual suggestions and thread creation messages, we already handled the thread creation message. What is left now?

Reply:
See, your logic is just the reverse. You do not actually want to check "is not a thread channel" but instead "is a message/guild/text channel" or whatever that channel is. I.e. something that exclusively identifies the channel you are interested in.

https://ci.dv8tion.net/job/JDA5/javadoc/net/dv8tion/jda/api/entities/ChannelType.html

Reply:
> is a message/guild/text channel

it is a message because MessageReceivedEvent only happens when it is a message
it is a guild channel because we already have the condition to check that inside the if statement
it is a text channel because messages can only be sent in a text channel + we know that suggestions channel is going to be a text channel

so now all we need to check is that it is not a thread inside the suggestions channel

hope that makes sense

Reply:
Sure that this can even happen? Due to the name-pattern it only listens to messages in this particular channel matching that name anyways.

Reply:
I just tried it out. It seems that the condition is obsolete. Thread messages do not enter the `onMessageReceived` due to the channel name pattern.

Reply:
hmm yes that can be the case i didn't though of that, so should i remove the if condition and test it again?

Reply:
yeah, thanks